plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java

```diff
@@ -328,9 +332,19 @@ private void reorderCapellaFragments(Scenario scenario, EList<EObject> textEleme
               // finish the Combined Fragment Block
               interactionFragments.add(capellaCombinedFragment.getFinish());
             }
-          }
+          } else if (elementFromXtext instanceof org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) {
+            if (((org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment) elementFromXtext).getKeyword()
```

Comment:
See comment below, no need to check here the "state", pass the keyword as param to getCorrespondingCapellaSteteFragment.